src/common/dns_resolve.cc

```diff
@@ -212,19 +212,19 @@ int DNSResolver::resolve_ip_addr(CephContext *cct, const string& hostname,
 int DNSResolver::resolve_ip_addr(CephContext *cct, res_state *res, const string& hostname, 
     entity_addr_t *addr) {
 
-  u_char nsbuf[NS_PACKETSZ];
+  std::unique_ptr<std::array<u_char, NS_MAXMSG>> nsbuf = std::make_unique<std::array<u_char, NS_MAXMSG>>();
```

Comment:
```suggestion
  auto nsbuf = std::make_unique<std::array<u_char, NS_MAXMSG>>();
```

might want to use `auto` here, as the type of `nsbuf` is self-explained with the `std::make_unique<..>()` statement.